torchtext/datasets/iwslt2016.py

```diff
@@ -225,50 +231,60 @@ def IWSLT2016(root='.data', split=('train', 'valid', 'test'), language_pair=('de
     src_eval, tgt_eval = valid_filenames
     src_test, tgt_test = test_filenames
 
-    extracted_files = []  # list of paths to the extracted files
-    dataset_tar = download_from_url(SUPPORTED_DATASETS['URL'], root=root, hash_value=SUPPORTED_DATASETS['MD5'],
-                                    path=os.path.join(root, SUPPORTED_DATASETS['_PATH']), hash_type='md5')
-    extracted_dataset_tar = extract_archive(dataset_tar)
-    # IWSLT dataset's url downloads a multilingual tgz.
-    # We need to take an extra step to pick out the specific language pair from it.
+    url_dp = IterableWrapper([URL])
+    cache_compressed_dp = url_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, _PATH),
+        hash_dict={os.path.join(root, _PATH): MD5},
+        hash_type="md5"
+    )
+    cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+    cache_compressed_dp = FileOpener(cache_compressed_dp, mode="b")
```

Comment:
Can we move this line after line 249 (after calling `on_disk_cache`) to be in line with our other implemenations (i.e. [EnWik9](https://github.com/pytorch/text/pull/1512/files#diff-6f50d62a0a75985ae909dd7f4ba8d66f9ec7091d8bb374ddacb8937cc5c27a72R44-R50)). Also we should call this `cache_decompressed_dp`.

So this can look like
```
...
cache_decompressed_dp = cache_compressed_dp.on_disk_cache(
        filepath_fn=lambda x: os.path.join(os.path.splitext(x[0])[0], iwslt_tar)
    )
cache_compressed_dp = FileOpener(cache_compressed_dp, mode="b").read_from_tar()
...
```